packages/teams-js/test/public/pages.spec.ts

```diff
@@ -210,42 +335,917 @@ describe('AppSDK-TeamsAPIs', () => {
+      });
+    });
+
+    describe('Testing pages.backStack.navigateBack function', () => {
+      it('pages.backStack.navigateBack should not allow calls before initialization', async () => {
+        await expect(pages.backStack.navigateBack()).rejects.toThrowError('The library has not yet been initialized');
+      });
+
+      Object.values(FrameContexts).forEach(context => {
+        it(`pages.backStack.navigateBack should register the navigateBack action when initialized with ${context} context`, () => {
+          utils.initializeWithContext(context);
+          pages.backStack.navigateBack();
+          const navigateBackMessage = utils.findMessageByFunc('navigateBack');
+          expect(navigateBackMessage).not.toBeNull();
+        });
+      });
+    });
+
+    describe('Testing pages.backStack.registerBackButtonHandler function', () => {
+      Object.values(FrameContexts).forEach(context => {
+        it('pages.backStack.registerBackButtonHandler should successfully register a back button handler and not call navigateBack if it returns true', async () => {
+          await utils.initializeWithContext(context);
+
+          let handlerInvoked = false;
+          pages.backStack.registerBackButtonHandler(() => {
+            handlerInvoked = true;
+            return true;
+          });
+
+          utils.sendMessage('backButtonPress');
+
+          let navigateBackMessage = utils.findMessageByFunc('navigateBack');
+          expect(navigateBackMessage).toBeNull();
+          expect(handlerInvoked).toBe(true);
+        });
+
+        it('pages.backStack.registerBackButtonHandler should successfully register a back button handler and call navigateBack if it returns false', async () => {
+          await utils.initializeWithContext(context);
+
+          let handlerInvoked = false;
+          pages.backStack.registerBackButtonHandler(() => {
+            handlerInvoked = true;
+            return false;
+          });
+
+          utils.sendMessage('backButtonPress');
+
+          let navigateBackMessage = utils.findMessageByFunc('navigateBack');
+          expect(navigateBackMessage).not.toBeNull();
+          expect(handlerInvoked).toBe(true);
+        });
+      });
+    });
+
+    describe('Testing pages.backStack.isSupported function', () => {
+      it('pages.backStack.isSupported should return false if the runtime says pages.backStack is not supported', () => {
+        utils.setRuntimeConfig({ apiVersion: 1, supports: { pages: {} } });
+        expect(pages.backStack.isSupported()).not.toBeTruthy();
+      });
+
+      it('pages.backStack.isSupported should return true if the runtime says pages.backStack is supported', () => {
+        utils.setRuntimeConfig({
+          apiVersion: 1,
+          supports: { pages: { backStack: {} } },
+        });
+        expect(pages.backStack.isSupported()).toBeTruthy();
+      });
+    });
+  });
+
+  describe('Testing pages.fullTrust namespace', () => {
```

Comment:
I also just noticed that we have test/private/pages.spec.ts where the pages.fullTrust capability appears to have tests. Can you also please reconcile these tests with that file?

Reply:
Do I need to remove those test cases in `test/private/pages.spec.ts` file?